fv3core/stencils/fv_dynamics.py

```diff
@@ -160,7 +160,7 @@ def compute_preamble(state, comm):
         )
 
 
-def do_dyn(state, comm):
+def do_dyn(state, comm, timer=None):
```

Comment:
if timer is None, please set it to a default value of `fv3gfs.util.NullTimer` (which we should make public), so you don't need repeated "if timer is not None" logic below. Then please use the `with` syntax for timings below.